### PR TITLE
[Workflow] Deprecated DefinitionBuilder::setInitialPlace()

### PR DESCRIPTION
**Pull request by lyrixx**
Added missing part of #30468

| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Workflow/DefinitionBuilder.php [hunk 1]
```diff
@@ -54,20 +54,36 @@ public function clear()
...
         $this->transitions = [];
-        $this->initialPlace = null;
+        $this->initialPlaces = null;
         $this->metadataStore = null;
 
         return $this;
     }
 
     /**
+     * @deprecated since Symfony 4.3. Use the setInitialPlaces() instead.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/DefinitionBuilder.php]: the ... method, or "the" should be removed

File: src/Symfony/Component/Workflow/DefinitionBuilder.php [hunk 2]
```diff
@@ -54,20 +54,36 @@ public function clear()
...
+     *
      * @param string $place
      *
      * @return $this
      */
     public function setInitialPlace($place)
     {
-        $this->initialPlace = $place;
+        @trigger_error(sprintf('Calling %s::setInitialPlace() is deprecated. Call %s::setInitialPlaces() instead.', __CLASS__, __CLASS__));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/DefinitionBuilder.php]: missing "since Symfony 4.3" etc - I'd suggest removing the repeated FQCN

**Review by nicolas-grekas — COMMENTED** [review]
I started reviewing this, but I'm not sure it's desired, the new name is not better when only one place exists - and code exists that would need to be upgraded, to a name that wouldn't fit well.
Since it's not obvious, it should maybe be for 4.4, if we still want to do it.

**Comment by lyrixx** [comment]
> I started reviewing this, but I'm not sure it's desired, the new name is not better when only one place exists - and code exists that would need to be upgraded, to a name that wouldn't fit well.

Everywhere else we are using `initialPlaces` - at the plural form.  So it's consistent with the rest of the component.

We should merge it in 4.3 to avoid confusion

File: src/Symfony/Component/Workflow/DefinitionBuilder.php (hunk 2 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: It's already in the annotation. and about the FQCN, I took the same message from https://github.com/symfony/symfony/blob/4.3/src/Symfony/Component/Workflow/Definition.php#L50-L64

File: src/Symfony/Component/Workflow/DefinitionBuilder.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: Fixed. And also `Definition::getInitialPlace()`

File: src/Symfony/Component/Workflow/DefinitionBuilder.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: That's a must: deprecation reports must be crystal clear, all runtime deprecations should report the affected package, Symfony or not. If some don't, they should be fixed.

Comment (lyrixx) [reply src/Symfony/Component/Workflow/DefinitionBuilder.php]: Fixed. I also fixed it in the definition class. I checked, everywhere else in the component it's OK.
I removed the double FQCN in both class

File: src/Symfony/Component/Workflow/Definition.php [hunk 3]
```diff
@@ -48,13 +48,13 @@ public function __construct(array $places, array $transitions, $initialPlaces =
...
     /**
-     * @deprecated since Symfony 4.3. Use the getInitialPlaces() instead.
+     * @deprecated since Symfony 4.3. Use getInitialPlaces() instead.
      *
      * @return string|null
      */
     public function getInitialPlace()
     {
-        @trigger_error(sprintf('Calling %s::getInitialPlace() is deprecated. Call %s::getInitialPlaces() instead.', __CLASS__, __CLASS__));
+        @trigger_error(sprintf('Calling %s::getInitialPlace() is deprecated since Symfony 4.3. Call getInitialPlaces() instead.', __CLASS__));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/Definition.php]: missing E_USER_DEPRECATED! (same below)

Comment (lyrixx) [reply src/Symfony/Component/Workflow/Definition.php]: Thanks, Fixed everywhere :)

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**